examples/keygen/keygen.c

```diff
@@ -41,9 +48,53 @@ static void usage(void)
     printf("./examples/keygen/keygen [keyblob.bin] [-ecc/-rsa/-sym] [-t] [-aes/xor]\n");
     printf("* -rsa: Use RSA for asymmetric key generation (DEFAULT)\n");
     printf("* -ecc: Use ECC for assymetric key generation \n");
-    printf("* -sym: Use AES Symmetric Cypher for key generation\n");
+    printf("* -sym: Use Symmetric Cypher for key generation\n");
+    printf("\tDefault Symmetric Cypher is AES CTR with 256 bits\n");
     printf("* -t: Use default template (otherwise AIK)\n");
     printf("* -aes/xor: Use Parameter Encryption\n");
+    printf("Example usage:\n");
+    printf("\t* RSA, default template\n");
+    printf("\t\t keygen -t\n");
+    printf("\t* ECC, Attestation Key template "\
+           "with AES CFB parameter encryption\n");
+    printf("\t\t keygen -ecc -aes\n");
+    printf("\t* Symmetric key, AES, CTR mode, 128 bits\n");
+    printf("\t\t keygen -sym=aesctr128\n");
+    printf("\t* Symmetric key, AES, CFB mode, 256 bits\n");
+    printf("\t\t keygen -sym=aescfb256\n");
+    printf("\t* Symmetric key, AES, CBC mode, 128 bits, "\
+           "with XOR parameter encryption\n");
+    printf("\t\t keygen -sym=aescbc256 -xor\n");
+}
+
+static int symChoice(const char* arg, TPM_ALG_ID* algSym, int* keyBits,
+                     char* symMode)
+{
+    if (XSTRNCMP(&arg[SYM_EXTRA_OPTS_POS+1], "aes", 3)) {
```

Comment:
Please add a check for length of `arg` before making assumptions about the size of it.

Reply:
Good point. I did a check in the caller. And I will add one here also.